src/value.rs

```diff
@@ -321,15 +319,18 @@ mod tests {
                     ''${ interpolation was escaped }
                     two single quotes: '''
                     three single quotes: ''''
-                "#.replace("|trailing-whitespace", "").as_str()
-            )),
-            vec![
-                StrPart::Literal(String::from(
-                    // Get the below with nix repl
-                    "    \n          \nThis is a multiline string :D\n  indented by two\n\\'\\'\\'\\'\\\n${ interpolation was escaped }\ntwo single quotes: ''\nthree single quotes: '''\n"
-                ))
-            ]
-        );
+                "#
+            .replace("|trailing-whitespace", "")
+            .as_str(),
+        ))[..]
+        {
```

Comment:
I think that inlining this in an if-condition is somewhat hard to read. Could you split this up into a declaration and a statement, please?